packages/x-scheduler/src/material/internals/components/day-time-grid/DayGridCell.tsx

```diff
@@ -1,20 +1,56 @@
 'use client';
 import * as React from 'react';
+import { useStore } from '@base-ui-components/utils/store';
 import { DayGrid } from '../../../../primitives/day-grid';
 import { useAdapter } from '../../../../primitives/utils/adapter/useAdapter';
 import { isWeekend } from '../../../../primitives/utils/date-utils';
 import { useEventOccurrencesWithDayGridPosition } from '../../../../primitives/use-event-occurrences-with-day-grid-position';
 import { EventPopoverTrigger } from '../event-popover';
 import { DayGridEvent } from '../event';
+import { useEventCalendarStoreContext } from '../../../../primitives/utils/useEventCalendarStoreContext';
+import { selectors } from '../../../../primitives/use-event-calendar';
+
 import './DayTimeGrid.css';
+import { useEventPopover } from '../event-popover/EventPopoverContext';
 
 export function DayGridCell(props: DayGridCellProps) {
   const { day, row } = props;
   const adapter = useAdapter();
   const placeholder = DayGrid.usePlaceholderInDay(day.value, row);
+  const store = useEventCalendarStoreContext();
+  const cellRef = React.useRef<HTMLDivElement | null>(null);
+  const rawPlaceholder = useStore(store, selectors.occurrencePlaceholder);
+  const { startEditing } = useEventPopover();
+
+  const onDoubleClick = () => {
+    store.setOccurrencePlaceholder({
+      eventId: null,
+      occurrenceKey: 'create-placeholder',
+      surfaceType: 'day-grid',
+      start: adapter.startOfDay(day.value),
+      end: adapter.endOfDay(day.value),
+      originalStart: null,
+    });
+  };
+
+  React.useEffect(() => {
+    if (!placeholder || !rawPlaceholder) {
+      return;
+    }
+
+    const isCreation = rawPlaceholder.eventId == null && rawPlaceholder.surfaceType === 'day-grid';
+    if (!isCreation) {
+      return;
+    }
+    startEditing({ currentTarget: cellRef.current } as unknown as React.MouseEvent, {
```

Comment:
😆 I think you can tweak the typing of the 1st param of `startEditing` to avoid creating an object like that. Maybe directly pass `currentTarget`.